freva-client/src/freva_client/query.py

```diff
@@ -712,103 +745,132 @@ def _facet_search(
             constraints = data["primary_facets"]
         return {f: v for f, v in data["facets"].items() if f in constraints}
 
-    def add_user_data(
-        self, username: str, paths: List[str], facets: Dict[str, str]
+    def userdata(
+        self,
+        action: Literal["add", "delete"],
+        userdata_items: List[Union[str, xr.Dataset]] = [],
+        metadata: Dict[str, str] = {},
```

Comment:
This can have very confusing consequences for the users. Do you assign these mutable objects in the function signature on purpose? If so rename them to _userdata_items, _metadata to show that this is internal. Otherwise, assign them to None and do a `userdata_times = userdata_items or []` etc. later.